### PR TITLE
14 get posts get poststag=memesanimal=cat filter posts by date

### PR DESCRIPTION
**Pull request by mohmmed23**

### CODE REVIEW

**Review by aaamra — COMMENTED** [review]
please add the testes

File: tsconfig.json [hunk 1]
```diff
@@ -32,5 +33,5 @@
   "include": [
       "src",
       "client"
```
Comment (aaamra) [inline tsconfig.json:35]: remove this line

File: src/controllers/GetPosts.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            [Op.substring]: searchCountent || ''
+          },
+          TagId: tagId
+        }
+
+        : filterData = {
+          content: {
+            [Op.substring]: ''
+          }
+        }
```
Comment (aaamra) [inline src/controllers/GetPosts.ts:37]: you are using nested ternary operators
this makes the code unreadable

use the `if`, `else if` and `else` 

and i want from you to add [this rule](https://eslint.org/docs/rules/no-nested-ternary) to `.eslintrc.json`

and you can look at [this article](https://medium.com/@benlmsc/stop-using-nested-ternary-operators-heres-why-53e7e078e65a)  to know more about this problem

File: src/controllers/GetPosts.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  console.log(filterData)
+
+  // : searchContent
+  // ? filterData = {
+  //   content: {
+  //     [Op.substring]: searchContent
+  //   }
+  // }
```
Comment (aaamra) [inline src/controllers/GetPosts.ts:45]: to push your commented lines

**Review by aaamra — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 1 above)
Comment (aaamra) [reply tsconfig.json:35]: i mean the client

File: src/controllers/GetPosts.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    filterData = {
+      content: {
+        [Op.substring]: searchCountent || ''
+      },
+      TagId: tagId
+    }
+  } else {
+    filterData = {
+      content: {
+        [Op.substring]: ''
```
Comment (aaamra) [inline src/controllers/GetPosts.ts:35]: there is a bug here
if there is no `tagId` or `animalId` you put the search to empty string
that means get all the data even if there is something in the `searchContent` 

and there is no test case for this situation

File: src/controllers/GetPosts.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import { Response, Request } from 'express'
+import { User, Post, Like, Tag, Animal } from '../db'
+import { Op } from 'sequelize'
+
+const getPosts = async (req: Request, res: Response) => {
+  const { tagId, animalId, searchCountent } = req.query
```
Comment (aaamra) [inline src/controllers/GetPosts.ts:6]: it's a common convention to use `q` in query string for searching

```suggestion
  const { tagId, animalId, q } = req.query
```

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by aaamra — APPROVED** [review]

File: src/db/build.ts [hunk 6]
```diff
@@ -1,14 +1,15 @@
...
-
+import data from './seeds.json'
 const buildDB = async () => {
   await sequelize.sync({ force: true })
   await Animal.bulkCreate(data.Animal)
   await Tag.bulkCreate(data.Tag)
   await User.bulkCreate(data.User)
   await Post.bulkCreate(data.Post)
   await Like.bulkCreate(data.Like)
+  const postd = await Post.findAll({})
```
Comment (AseelL) [inline src/db/build.ts]: remove the logging from here

**Review by AseelL — CHANGES_REQUESTED** [review]

**Review by AseelL — APPROVED** [review]

**Review by Saeed99Madi — APPROVED** [review]

File: __test__/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
+import { describe, expect, test } from '@jest/globals'
+import app from '../src/app'
+import supertest from 'supertest'
+import build from '../src/db/build'
+import sequelize from '../src/db/connection'
+
+beforeEach(() => build())
+
+describe('all post ', () => {
```
Comment (AseelL) [inline __test__/index.test.ts:9]: no need to rewrite describe for each test, write a single describe for the router and give each test a descriptive name

**Review by AseelL — CHANGES_REQUESTED** [review]

**Review by AseelL — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Review by Saeed99Madi — APPROVED** [review]

**Review by salsabeelomar — APPROVED** [review]

**Pull request merged**